pkg/endpoint/endpoint.go

```diff
@@ -2387,6 +2387,15 @@ func (e *Endpoint) syncPolicyMap() error {
 		e.realizedMapState = make(map[policymap.PolicyKey]struct{})
 	}
 
+	if e.desiredMapState == nil {
+		e.desiredMapState = make(map[policymap.PolicyKey]struct{})
+	}
+
+	if e.PolicyMap == nil {
+		e.getLogger().Debug("not syncing PolicyMap state for endpoint because PolicyMap is nil")
+		return nil
```

Comment:
We have a mechanism to exit the controller, if that's a simpler solution (https://github.com/cilium/cilium/blob/master/pkg/controller/controller.go#L32-L40). I don't think it would be that useful in any case.

Reply:
Oh, actually, it might be better to return the error here. That would record this information in the controller status. If it's transient, it will clear itself out, if it isn't then it would show up in the status output. The exit thing above also records the "reason" for exiting.